File: src/Tests/MimironsGoldOMatic.WoWAddon.Tests/run_tests.lua
--[[
  LuaUnit runner for WoW addon tests (requires Lua 5.1+ / LuaJIT / Lua 5.4).
  Usage: lua run_tests.lua
]]

local TEST_ROOT = "."
if arg and arg[0] and arg[0] ~= "" then
  local p = arg[0]
  local d = p:match("^(.*[/\\])")
  if d then
    TEST_ROOT = d
  end
end

package.path = TEST_ROOT .. "lib/?.lua;" .. package.path

CORE_LUA = TEST_ROOT .. "../MimironsGoldOMatic.WoWAddon/MimironsGoldOMatic.Core.lua"
ADDON_LUA = TEST_ROOT .. "../MimironsGoldOMatic.WoWAddon/MimironsGoldOMatic.lua"
MOCK_LUA = TEST_ROOT .. "wow_api_mock.lua"

local lu = require("luaunit")

dofile(TEST_ROOT .. "UnitTests/core_tests.lua")
dofile(TEST_ROOT .. "IntegrationTests/addon_integration_tests.lua")

os.exit(lu.LuaUnit.run())

File: src/Tests/MimironsGoldOMatic.WoWAddon.Tests/run_tests.py
#!/usr/bin/env python3
"""Run LuaUnit tests for MimironsGoldOMatic.WoWAddon (requires lua on PATH)."""
import os
import shutil
import subprocess
import sys


def main() -> int:
    os.chdir(os.path.dirname(os.path.abspath(__file__)))
    lua = shutil.which("lua")
    if not lua:
        print(
            "lua not found on PATH. Install Lua (e.g. choco install lua on Windows, "
            "apt install lua5.4 on Linux) or add lua to PATH.",
            file=sys.stderr,
        )
        return 1
    cmd = [lua, "run_tests.lua", *sys.argv[1:]]
    return subprocess.run(cmd, check=False).returncode


if __name__ == "__main__":
    raise SystemExit(main())
